tests/test_conversion_cyclical.py

```diff
@@ -0,0 +1,149 @@
+# Imports sorted alphabetically
+import pathlib
+import pytest
+
+# 'from' imports, sorted alphabetically by module
+from google.protobuf.json_format import Parse
+from google.protobuf.message import Message
+from protarrow.common import M, ProtarrowConfig
+from protarrow.proto_to_arrow import (
+    messages_to_record_batch,
+    ProtarrowCycleError,
+    messages_to_table,
+)
+from protarrow_protos.bench_pb2 import (
+    CyclicalDirectMessage,
+    CyclicalIndirectMessageLevel1,
+    CyclicalMapMessage,
+    CyclicalRepeatedMessage,
+)
+from typing import List, Type
+
+CONFIGS = [
+    ProtarrowConfig(purge_cyclical_messages=False),
+    ProtarrowConfig(purge_cyclical_messages=True),
+]
+DIR = pathlib.Path(__file__).parent
+
+
+def read_proto_jsonl(path: pathlib.Path, message_type: Type[M]) -> List[M]:
+    with path.open() as fp:
+        return [
+            Parse(line.strip(), message_type())
+            for line in fp
+            if line.strip() and not line.startswith("#")
+        ]
+
+
+def _load_data(filename: str, message_type: Type[Message]) -> List[Message]:
+    """Loads messages from the specific test data file."""
+    source_file = DIR / "data" / filename
+    source_messages = read_proto_jsonl(source_file, message_type)
+    if not source_messages:
+        pytest.skip(f"Test data file {filename} is empty or missing.")
+    return source_messages
+
+
+# ====================================================================
+# DIRECT SELF-REFERENCE
+#  X           X
+#  A - Y  =>   A
+#      A
+# ====================================================================
+@pytest.mark.parametrize("config", CONFIGS)
+def test_cyclical_direct_message_handling(config: ProtarrowConfig):
+    messages = _load_data("CyclicalDirectMessage.jsonl", CyclicalDirectMessage)
+
+    if not config.purge_cyclical_messages:
+        with pytest.raises(ProtarrowCycleError):
```

Comment:
can you test the execption message with with `pytest.raises(ProtarrowCycleError, match=r"some regex"):`

In particular it's important that we check the trace of messages name is present (it will be valuable for the user)